allensdk/brain_observatory/behavior/stimulus_processing/stimulus_templates.py

```diff
@@ -0,0 +1,107 @@
+from typing import Dict, List
+
+import numpy as np
+import pandas as pd
+
+from allensdk.brain_observatory.behavior.stimulus_processing.util import \
+    get_image_set_name, convert_filepath_caseinsensitive
+
+
+class StimulusImage(np.ndarray):
```

Comment:
did you consider using an established class for what you want here?
https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.info